src/vhip/mlmodel/gene_features.py

```diff
@@ -352,3 +354,38 @@ def RSCU(
                         synonymous_total_count / len(synonymous_codons)
                     )  # expected frequency of current codon given all assumption all synonymous codons are equally likely to encode the aa
                     self.RSCU_dict[codon] = self.codon_dict[codon] / expected_frequency
+
+
+class CodonBiasComparison:
+    """Class for calculating codon bias similarity between a virus GeneSet and a host GeneSet.
+
+    Args:
+        host_dict: Dictionary of codons/amino acids and their counts/frequencies/RSCUs in a host GeneSet.
+        virus_dict: Dictionary of codons/amino acids and their counts/frequencies/RSCUs in a virus GeneSet.
+    """
+
+    def __init__(
+        self,
+        host_dict: Union[dict[str, int], dict[str, float]],
+        virus_dict: Union[dict[str, int], dict[str, float]],
+    ) -> None:
+        """Initialize class variables and read in an annotated genes file, storing Gene objects and metadata in lists."""
+        self.host_dict = host_dict
+        self.host_list = list(self.host_dict.values())
+        self.virus_dict = virus_dict
+        self.virus_list = list(self.virus_dict.values())
+
+    def slope(self) -> None:
+        """Compute slope between host and virus codon bias using linear regression."""
+        self.slope = scipy.stats.linregress(self.host_list, self.virus_list)[0]
```

Comment:
can you add a comment of why you keep the first value?

Reply:
added!